parser.py

```diff
@@ -142,7 +160,36 @@ def __init__(self, condition, op):
     def __repr__(self):
         return f"if ({self.condition}) {self.op}"
 
-# TODO: make separate class for classical operations like `measure ...`
+
+class TensorOp:
+    def __init__(self, *ops):
+        self.ops = ops
+
+    @property
+    def params(self):
+        p = []
+        for o in self.ops:
+            p.extend(o.params)
+        return p
+
+    @property
+    def wires(self):
+        w = []
+        for o in self.ops:
+            w.extend(o.wires)
+        return w
```

Comment:
These would be better created on calls to `__init__`, no?

Reply:
Yeah. I did it this way because I thought originally that the tensor might have to be constantly mutated by appending new ops during parsing.

But instead, it ended up just creating a nested tensor object:
```
Tensor[op1, Tensor[op2, Tensor[op3, op4]]]
```
which works fine as well (and maybe a better approach). So we can move this into `__init__`